sycl/include/sycl/queue.hpp

```diff
@@ -245,6 +247,183 @@ class __SYCL_EXPORT queue : public detail::OwnerLessBase<queue> {
   queue(const context &SyclContext, const device &SyclDevice,
         const async_handler &AsyncHandler, const property_list &PropList = {});
 
+#else // __SYCL_EXT_ONEAPI_BACKEND_LEVEL_ZERO_V3
+
+private:
```

Comment:
Why can't we use inline versioning namespace for this purpose? IMO, that would be way cleaner...

Reply:
This is a temporary measure for distinguishing calls to queue constructors from pre-compiled binaries versus current binaries. The distinction needs to be made to use different defaults for immediate command lists. This was a hard requirement from the oneAPI PXT. In 2024.0, with ABI break, we will remove this requirement and this feature.
This solution was necessary because all other more subtle attempts failed to properly distinguish calls on Windows. On Linux, simpler solutions worked.